Config.php

```diff
@@ -67,12 +67,48 @@ public function __construct(\PDO $dbh, $config_source = NULL, $config_type = '',
                 break;
             }
             case 'json': {
+                // check valid keys
+                if (empty($config_source)) die('PHPAuth: config type is FILE, but no source file declared!'); //@todo: \Exception
+
+                // replace beginner '$' in filepath to application root directory
+                $source = preg_replace('/^\$/', getcwd(), $config_source);
+
+                // check json-config is readable
+                if (!is_readable($source)) die("PHPAuth: config type is FILE, declared as {$source}, but file not readable or not exist"); //@todo: \Exception
+
+                // load configuration
+                $this->config = json_decode(file_get_content($source));
```

Comment:
It's useful to check for JSON errors, I think. 
https://www.php.net/manual/en/function.json-last-error.php